### PR TITLE
add cni config validation before loading network

### PR DESCRIPTION
**Pull request by mars1024**
ref to #49 

Signed-off-by: Bruce Ma <brucema19901024@gmail.com>

### CODE REVIEW

**Comment by mars1024** [comment]
All the test failures have been resolved, could you help to go on reviewing this? @dcbw @rajatchopra  cc @squeed @mrunalp

File: pkg/ocicni/ocicni.go [hunk 1]
```diff
@@ -473,6 +482,16 @@ func (plugin *cniNetworkPlugin) GetPodNetworkStatus(podNetwork PodNetwork) ([]cn
 	return results, nil
 }
 
+func (network *cniNetwork) validateNetwork() error {
+	netConf, cniNet := network.NetworkConfig, network.CNIConfig
+
+	if _, err := cniNet.ValidateNetworkList(context.TODO(), netConf); err != nil {
+		return fmt.Errorf("cni config %s validation fails : %v", network.name, err)
```
Comment (saschagrunert) [inline pkg/ocicni/ocicni.go]: Just a nit:
```suggestion
		return fmt.Errorf("CNI config %q validation fails: %v", network.name, err)
```

Comment (mars1024) [reply pkg/ocicni/ocicni.go]: Thanks, done!

**Comment by squeed** [comment]
as a CNI maintainer (and the author of the Validate() function,
/lgtm

**Comment by haircommander** [comment]
/lgtm
Will need @dcbw or another approver to approve

**Pull request merged**